packages/runtime/runtime-utils/src/serializer.ts

```diff
@@ -32,92 +36,153 @@ export class FluidSerializer implements IFluidSerializer {
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+        return !!input && typeof input === "object"
+            ? this.recursivelyDecode(input)
+            : input;
+    }
+    */
+
     public stringify(input: any, bind: IFluidHandle) {
         return JSON.stringify(input, (key, value) => {
-            // If the current 'value' is not a handle, return it unmodified.  Otherwise,
-            // return the result of 'serializeHandle'.
             // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-            const handle = !!value && value.IFluidHandle;
-            // TODO - understand why handle === false in some of our tests
-            // eslint-disable-next-line max-len
-            // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions,@typescript-eslint/no-unsafe-return
-            return handle
-                ? this.serializeHandle(handle, bind)
-                : value;
+            if (value) {
+                // If 'value' is an IFluidHandle return its serialized form.
+                const handle = value.IFluidHandle;
+                if (handle !== undefined) {
+                    return this.serializeHandle(handle, bind);
+                }
+            }
+            return value;
         });
     }
 
     // Parses the serialized data - context must match the context with which the JSON was stringified
     public parse(input: string) {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
         return JSON.parse(
             input,
             (key, value) => {
-                if (!isSerializedHandle(value)) {
-                    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-                    return value;
-                }
+                // If 'value' is a serialized IFluidHandle return the deserialized result.
+                if (isSerializedHandle(value)) {
+                    // Old documents may have handles with relative path in their summaries. Convert these to absolute
+                    // paths. This will ensure that future summaries will have absolute paths for these handles.
+                    const absolutePath = value.url.startsWith("/")
+                        ? value.url
+                        : generateHandleContextPath(value.url, this.context);
 
-                // Old documents may have handles with relative path in their summaries. Convert these to absolute
-                // paths. This will ensure that future summaries will have absolute paths for these handles.
-                const absolutePath = value.url.startsWith("/")
-                    ? value.url
-                    : generateHandleContextPath(value.url, this.context);
-                return new RemoteFluidObjectHandle(absolutePath, this.root);
+                    return new RemoteFluidObjectHandle(absolutePath, this.root);
+                }
+                return value;
             });
```

Comment:
The body of the 'parse()' lambda is now emitted by the same codegen template that is used for 'recursivelyDecode()'.  (Previously, the implementations could drift.)